File: app/src/main/java/io/dwak/holohackernews/app/dagger/component/AppComponent.java
package io.dwak.holohackernews.app.dagger.component;

import android.app.Application;

import javax.inject.Named;

import dagger.Component;
import io.dwak.holohackernews.app.cache.CacheManager;
import io.dwak.holohackernews.app.dagger.module.AppModule;

@Component(modules = {AppModule.class})
public interface AppComponent {
    void inject(Application application);
    @Named("cacheManager")
    CacheManager getCacheManager();
}

File: app/src/main/kotlin/io/dwak/holohackernews/app/dagger/component/AppComponent.kt
package io.dwak.holohackernews.app.dagger.component

import android.content.Context
import dagger.Component
import io.dwak.holohackernews.app.HackerNewsApplication
import io.dwak.holohackernews.app.dagger.module.AppModule
import io.dwak.holohackernews.app.ui.list.presenter.StoryItemPresenterImpl

@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun inject(storyItemPresenter : StoryItemPresenterImpl)

    val app : HackerNewsApplication
}